packages/jaeger-ui/src/components/TraceDiff/TraceDiffHeader/TraceHeader.test.js

```diff
@@ -13,57 +13,75 @@
 // limitations under the License.
 
 import * as React from 'react';
-import { shallow } from 'enzyme';
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
 
 import TraceHeader, { Attrs, EmptyAttrs } from './TraceHeader';
 import { fetchedState } from '../../../constants';
 
 describe('TraceHeader', () => {
-  const props = {
-    duration: 700,
-    error: { errorKey: 'errorValue' },
-    traceID: 'trace-id',
-    traceName: 'trace name',
-  };
-  let wrapper;
+  const renderWithProps = (passedProps = {}) => {
+    const originalProps = {
+      duration: 700,
+      error: { errorKey: 'errorValue' },
+      traceID: 'trace-id',
+      traceName: 'trace name',
+    };
 
-  beforeEach(() => {
-    wrapper = shallow(<TraceHeader {...props} />);
-  });
+    const props = {
+      ...originalProps,
+      ...passedProps,
+    };
+
+    render(<TraceHeader {...props} />);
+  };
 
   it('renders as expected', () => {
-    expect(wrapper).toMatchSnapshot();
+    renderWithProps();
+    expect(screen.getAllByTestId('TraceDiffHeader--traceHeader').length).toBe(1);
   });
 
-  it('renders populated attrs component when props.state === fetchedState.DONE', () => {
-    wrapper.setProps({
+  it('renders populated Attrs component when props.state === fetchedState.DONE', () => {
+    renderWithProps({
       startTime: 150,
       totalSpans: 50,
       state: fetchedState.DONE,
     });
-    expect(wrapper).toMatchSnapshot();
+    expect(screen.getAllByTestId('TraceDiffHeader--traceAttr').length).toBe(3);
   });
 
   it('renders "Select a Trace..." when props.traceID is not provided ', () => {
-    wrapper.setProps({
+    renderWithProps({
       traceID: null,
     });
-    expect(wrapper.find('.u-tx-muted').text()).toBe('Select a Trace...');
+    expect(screen.getByText('Select a Trace...'));
   });
 
   describe('EmptyAttrs', () => {
     it('renders as expected', () => {
-      expect(shallow(<EmptyAttrs />)).toMatchSnapshot();
+      render(<EmptyAttrs />);
+      expect(screen.getAllByTestId('TraceDiffHeader--traceAttr').length).toBe(1);
+      expect(screen.getByTestId('TraceDiffHeader--traceAttr').textContent.trim()).toBe('');
     });
   });
 
   describe('Attrs', () => {
     it('renders as expected when provided props', () => {
-      expect(shallow(<Attrs duration={700} startTime={150} totalSpans={50} />)).toMatchSnapshot();
+      render(<Attrs duration={700} startTime={150} totalSpans={50} />);
+
+      // Test that the shown values are correctly formatted
+      expect(screen.getByText('January 1, 1970, 12:00:00 am'));
```

Comment:
1. no assertion
2. where is this exact date coming from? Could we get something more meaningful & matching the inputs?

Reply:
> getByText is itself an assertion

why, doesn't it return some empty value if it cannot find? or does it throw exception?

> We are providing 150 in the startTime attribute

I believe we typically represent times/durations in microseconds, for timestamps specifically microseconds since epoch. So when you have 150us since epoch rendered as just a datetime with seconds precision you see no difference since the epoch itself. If it's 450us instead of 150us - the output would be the same. So ideally we should use some large enough number, e.g. 1,000,000us, which the will at least change the epoch timestamp to `12:00:01 am`

Reply:
1. `getBy` queries return exactly one element matching from the DOM, and throw an exception if no element is found or if multiple elements are found.
2. Makes sense. Will incorporate the same in tests and add comments about it.